components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/dao/impl/DeviceOrganizationDAOImpl.java

```diff
@@ -0,0 +1,456 @@
+            stmt.setString(1, deviceId);
+            rs = stmt.executeQuery();
+            if (rs.next()) {
+                isGatewayState = rs.getInt("IS_GATEWAY");
+            }
+        } catch (SQLException e) {
+            throw new DeviceOrganizationDAOException("Error occurred for device with ID: " + deviceId +
+                    " while checking if device is a gateway.", e);
+        } finally {
+            DeviceManagementDAOUtil.cleanupResources(stmt, rs);
+            return isGatewayState;
+        }
+    }
+
+    /**
+     * This method allows to get the Children connected to a parent
+     *
+     * @param parentId unique device identifier, in this case the parents' ID
+     * @return DeviceOrganizationMetadataHolder ArrayList with IDs of children
+     * @throws DeviceOrganizationDAOException
+     */
+    @Override
+    public List<DeviceOrganizationMetadataHolder> getChildrenByParentId(String parentId) throws DeviceOrganizationDAOException {
+        List<DeviceOrganizationMetadataHolder> children = new ArrayList<>();
+        Connection conn;
+        PreparedStatement stmt = null;
+        ResultSet rs = null;
+        DeviceOrganizationMetadataHolder deviceMetadataHolder;
+        try {
+            conn = this.getConnection();
+            String sql = "SELECT * FROM DEVICE_ORGANIZATION_MAP WHERE DEVICE_PARENT = ?";
+            stmt = conn.prepareStatement(sql);
+            stmt.setString(1, parentId);
+            rs = stmt.executeQuery();
+            while (rs.next()) {
+                deviceMetadataHolder = this.loadOrganization(rs);
+                children.add(deviceMetadataHolder);
+            }
+        } catch (SQLException e) {
+            throw new DeviceOrganizationDAOException("Error occurred for device with ID: " + parentId +
+                    " while retrieving children.", e);
+        } finally {
+            DeviceManagementDAOUtil.cleanupResources(stmt, rs);
+            return children;
+        }
+    }
+
+    /**
+     * This method allows to retrieve the list of children IDs connected to a list of parents
+     *
+     * @param parentIds unique device identifiers, in this case the parents' IDs
+     * @return DeviceOrganizationMetadataHolder ArrayList with IDs of children
+     * @throws DeviceOrganizationDAOException
+     */
+    @Override
+    public List<DeviceOrganizationMetadataHolder> getChildrenByParentId(List<String> parentIds) throws
+            DeviceOrganizationDAOException {
+        List<DeviceOrganizationMetadataHolder> children = new ArrayList<>();
+        Connection conn;
+        PreparedStatement stmt = null;
+        ResultSet rs = null;
+        DeviceOrganizationMetadataHolder deviceMetadataHolder;
+        Object[] data = parentIds.toArray();
+        try {
+            conn = this.getConnection();
+            String sql = "SELECT * FROM DEVICE_ORGANIZATION_MAP WHERE DEVICE_PARENT IN (SELECT * FROM TABLE(x VARCHAR = ?))";
```

Comment:
Does this query works with MySQL? I guess this will only works with H2 database engine? Please check on this. Also please check on the line formatting too. This might exceeds 110 column length.